### PR TITLE
added option to override biostudies submission in order to make point fixes

### PR DESCRIPTION
**Pull request by sherwoodf**
Update to ingest to have an override option in order to make point fixes to studies.

This might be preferable to editing the study directly when is not a study that we are responsible for

To test, run: poetry run biaingest ingest --dryrun S-BIAD15 S-BIAD954 S-BIAD978 S-BIAD1076 S-BIAD1136 S-BIAD1223 S-BIAD1261 S-BIAD1344 S-BSMS4 S-BSST651 S-BSST744 S-BIAD1285 --write-csv output_table_w_filelist_05.csv --process-filelist skip

(note S-BIAD1285 has no overrides, just to make sure nothing was changed with normal running)

### CODE REVIEW

File: bia-ingest/bia_ingest/biostudies/api.py [hunk 1]
```diff
@@ -188,6 +188,39 @@ def load_submission_table_info(accession_id: str) -> SubmissionTable:
...
+    # Note this is a dictionary to include reasons why the override was made
+    overrides = {
+        "S-BSMS4": "Author links to affiliations were missing 'reference: true'",
+        "S-BIAD15": "Invalid licence, and affilicaiton was missing accno",
+        "S-BIAD1076": "Biosample had the Experimental variables text split up into 3 sections, possibly due to commas?",
+        "S-BIAD1261": "Author had no name, and the email was for a whole lab. Affiliation name was used instead of author name",
+        "S-BIAD978": "Unreferenced Image analysis-5, and a broken association to image analysis 1 (that doesn't exist)",
+        "S-BIAD954": "invalid email: Julia Nöth <julia.noeth@ufz.de> changed to: julia.noeth@ufz.de",
+        "S-BIAD1136": "invalid email: oona.paavolainen@ut changed to: oona.paavolainen@utu.fi (same ending as other authors)",
+        "S-BIAD1223": "invalid email: ylva.ivarsson@kemi..u.se changed to: ylva.ivarsson@kemi.u.se",
```
Comment (AybukeKY) [inline bia-ingest/bia_ingest/biostudies/api.py:200]: Quick google tells me this should be ylva.ivarsson@kemi.uu.se

File: bia-ingest/bia_ingest/biostudies/api.py [hunk 2]
```diff
@@ -188,6 +188,39 @@ def load_submission_table_info(accession_id: str) -> SubmissionTable:
...
 def load_submission(accession_id: str) -> Submission:
+    # Note this is a dictionary to include reasons why the override was made
+    overrides = {
+        "S-BSMS4": "Author links to affiliations were missing 'reference: true'",
+        "S-BIAD15": "Invalid licence, and affilicaiton was missing accno",
+        "S-BIAD1076": "Biosample had the Experimental variables text split up into 3 sections, possibly due to commas?",
+        "S-BIAD1261": "Author had no name, and the email was for a whole lab. Affiliation name was used instead of author name",
+        "S-BIAD978": "Unreferenced Image analysis-5, and a broken association to image analysis 1 (that doesn't exist)",
+        "S-BIAD954": "invalid email: Julia Nöth <julia.noeth@ufz.de> changed to: julia.noeth@ufz.de",
+        "S-BIAD1136": "invalid email: oona.paavolainen@ut changed to: oona.paavolainen@utu.fi (same ending as other authors)",
```
Comment (AybukeKY) [inline bia-ingest/bia_ingest/biostudies/api.py:199]: Internet tells me this should be oona.k.paavolainen@utu.fi

File: bia-ingest/submission_overrides/biostudies/S-BIAD1223/S-BIAD1223_override.json [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+          "reference" : true
+        } ]
+      }, {
+        "type" : "author",
+        "attributes" : [ {
+          "name" : "Name",
+          "value" : "Ylva Ivarsson"
+        }, {
+          "name" : "E-mail",
+          "value" : "ylva.ivarsson@kemi.u.se"
```
Comment (AybukeKY) [inline bia-ingest/submission_overrides/biostudies/S-BIAD1223/S-BIAD1223_override.json:73]: should be "ylva.ivarsson@kemi.uu.se"

File: bia-ingest/submission_overrides/biostudies/S-BIAD1136/S-BIAD1136_override.json [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        } ]
+      } ],
+      "subsections" : [ {
+        "type" : "author",
+        "attributes" : [ {
+          "name" : "Name",
+          "value" : "Oona Paavolainen"
+        }, {
+          "name" : "E-mail",
+          "value" : "oona.paavolainen@utu.fi"
```
Comment (AybukeKY) [inline bia-ingest/submission_overrides/biostudies/S-BIAD1136/S-BIAD1136_override.json:48]: should be "oona.k.paavolainen@utu.fi"

File: bia-ingest/submission_overrides/biostudies/S-BIAD1261/S-BIAD1261_override.json [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        "valqual" : [ {
+          "name" : "URL",
+          "value" : "https://creativecommons.org/publicdomain/zero/1.0/legalcode"
+        } ]
+      } ],
+      "subsections" : [ {
+        "type" : "author",
+        "attributes" : [ {
+          "name" : "Name",
+          "value" : "Ruhr University Bochum"
```
Comment (AybukeKY) [inline bia-ingest/submission_overrides/biostudies/S-BIAD1261/S-BIAD1261_override.json:42]: May be better to have name as "Cytology department" and affiliation as "Ruhr University Bochum".

**Review by AybukeKY — COMMENTED** [review]
Have some minor comments about changes, otherwise looks good.

File: bia-ingest/bia_ingest/biostudies/api.py (hunk 2 above)
Comment (sherwoodf) [reply bia-ingest/bia_ingest/biostudies/api.py:199]: This feels like more of a change - i dunno whether that's stepping over a line we're actually worried about

Comment (AybukeKY) [reply bia-ingest/bia_ingest/biostudies/api.py:199]: Good point; the other address may be wrong as we haven't checked so we can either omit the k as they're not the submitter and it's connected to their orcid id anyway, or we can notify them when we do the resubmission.

**Review by AybukeKY — APPROVED** [review]

**Pull request merged**